components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ServiceTest.java

```diff
@@ -172,4 +213,174 @@ public void testIdentityOAuth2Exception() throws Exception {
+        when(authenticatedUser.toString()).thenReturn("testAuthenticatedUser");
+
+        AccessTokenDO accessTokenDO = new AccessTokenDO();
+        accessTokenDO.setConsumerKey("testCosumerKey");
+        accessTokenDO.setAuthzUser(authenticatedUser);
+
+
+        TokenMgtDAO tokenMgtDAO = mock(TokenMgtDAO.class);    //mock methods for tokenMgtDAO
+        doNothing().when(tokenMgtDAO).revokeTokens(any(String[].class));
+        when(tokenMgtDAO.retrieveAccessToken(anyString(), anyBoolean())).thenReturn(accessTokenDO);
+        whenNew(TokenMgtDAO.class).withAnyArguments().thenReturn(tokenMgtDAO);
+
+
+        OAuthRevocationRequestDTO revokeRequestDTO = new OAuthRevocationRequestDTO();
+        revokeRequestDTO.setConsumerKey("testCosumerKey");
+        revokeRequestDTO.setToken("testToken");
+        revokeRequestDTO.setToken_type(GrantType.CLIENT_CREDENTIALS.toString());
+
+        when(oAuthCache.getValueFromCache(any(OAuthCacheKey.class))).thenReturn(accessTokenDO);
+        mockStatic(OAuthCache.class);
+        when(OAuthCache.getInstance()).thenReturn(oAuthCache);
+
+        oAuth2Service.revokeTokenByOAuthClient(revokeRequestDTO);
+        assertFalse(oAuth2Service.revokeTokenByOAuthClient(revokeRequestDTO).isError());
+
+//        when(oAuthCache.getValueFromCache(any(OAuthCacheKey.class))).thenReturn(null);
+//        assertFalse(oAuth2Service.revokeTokenByOAuthClient(revokeRequestDTO).isError());
+
+        when(OAuth2Util.authenticateClient(anyString(), anyString())).thenThrow(new InvalidOAuthClientException(" "));
+        assertEquals(oAuth2Service.revokeTokenByOAuthClient(revokeRequestDTO).getErrorMsg(), "Unauthorized Client");
+    }
+
+    @Test
+    public void testIdentityOAuth2ExceptionForRevokeTokenByOAuthClient() throws Exception {
+
+        when(oAuthEventInterceptorProxy.isEnabled()).thenReturn(true);
+        doThrow(new IdentityOAuth2Exception("")).when(oAuthEventInterceptorProxy)
+                .onPreTokenRevocationByClient(any(OAuthRevocationRequestDTO.class), anyMap());
+        when(oAuthComponentServiceHolder.getOAuthEventInterceptorProxy()).thenReturn(oAuthEventInterceptorProxy);
+        mockStatic(OAuthComponentServiceHolder.class);
+        when(OAuthComponentServiceHolder.getInstance()).thenReturn(oAuthComponentServiceHolder);
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(oAuthServerConfiguration.getPersistenceProcessor()).thenReturn(persistenceProcessor);
+        TokenMgtDAO tokenMgtDAO = mock(TokenMgtDAO.class);    //mock methods for tokenMgtDAO
+        whenNew(TokenMgtDAO.class).withAnyArguments().thenReturn(tokenMgtDAO);
+
+        OAuthRevocationRequestDTO revokeRequestDTO = new OAuthRevocationRequestDTO();
+        revokeRequestDTO.setConsumerKey("testCosumerKey");
+        revokeRequestDTO.setToken("testToken");
+        String result = "Error occurred while revoking authorization grant for applications";
```

Comment:
Do you really need this variable?

Reply:
I can replace it then it will create a long line